component/src/main/java/io/siddhi/extension/io/file/FileHandler.java

```diff
@@ -0,0 +1,456 @@
+                if (!fileList.contains(initialMapEntry)) {
+                    initialMap.remove(initialMapEntry.getAbsolutePath());
+                    sourceEventListener.onEvent(getFileHandlerEvent(initialMapEntry, listeningFileUri,
+                            STATUS_REMOVE), null);
+                }
+            }
+        }
+    }
+
+    /**
+     * Implementation of  FileAlterationImpl
+     */
+    public class FileAlterationImpl implements FileAlterationListener {
+
+        @Override
+        public void onStart(final FileAlterationObserver observer) {
+        }
+
+        @Override
+        public void onDirectoryCreate(final File directory) {
+            log.debug(directory.getAbsolutePath() + " was created.");
+            initialMap.put(directory.getAbsolutePath(), directory.length());
+            sourceEventListener.onEvent
+                    (getFileHandlerEvent(directory, listeningFileUri, STATUS_DONE), null);
+        }
+
+        @Override
+        public void onDirectoryChange(final File directory) {
+            log.debug(directory.getAbsolutePath() + " was modified");
+            synchronized (this) {
+                if (!recentFileSizeMap.containsKey(directory.getAbsolutePath())) {
+                    recentFileSizeMap.put(directory.getAbsolutePath(), directory.length());
+                    sourceEventListener.onEvent
+                            (getFileHandlerEvent(directory, listeningFileUri, STATUS_PROCESS),
+                                    null);
+                    ThreadOnContinuousChange threadOnContinuousChange = new ThreadOnContinuousChange(directory);
+                    siddhiAppContext.getScheduledExecutorService().schedule(threadOnContinuousChange,
+                            monitoringInterval, TimeUnit.MILLISECONDS);
+                    threadList.add(threadOnContinuousChange);
+                }
+            }
+        }
+
+        @Override
+        public void onDirectoryDelete(final File directory) {
+            log.debug(directory.getAbsolutePath() + " was deleted.");
+            initialMap.remove(directory.getAbsolutePath(), directory.length());
+            sourceEventListener.onEvent
+                    (getFileHandlerEvent(directory, listeningFileUri, STATUS_DONE), null);
+        }
+
+        @Override
+        public void onFileCreate(final File file) {
+            log.debug(file.getAbsolutePath() + " was created.");
+            initialMap.put(file.getAbsolutePath(), file.length());
+            sourceEventListener.onEvent
+                    (getFileHandlerEvent(file, listeningFileUri, STATUS_DONE), null);
+        }
+
+        @Override
+        public void onFileChange(final File file) {
+            log.debug(file.getAbsolutePath() + " was modified.");
+            synchronized (this) {
+                if (!recentFileSizeMap.containsKey(file.getAbsolutePath())) {
+                    recentFileSizeMap.put(file.getAbsolutePath(), file.length());
```

Comment:
seems we can use the same map used in the connect() and other methods here as well

Reply:
fixed in https://github.com/siddhi-io/siddhi-io-file/pull/82/commits/3f9a2a0eb50655821dd322688aa034be33ef8860